lightning/src/onion_message/messenger.rs

```diff
@@ -875,6 +941,17 @@ where
 		}
 		msgs
 	}
+
+	fn enqueue_event(&self, event: Event) {
+		const MAX_EVENTS_BUFFER_SIZE: usize = (1 << 10) * 256;
+		let mut pending_events = self.pending_events.lock().unwrap();
+		let total_buffered_bytes: usize = pending_events
+			.iter()
+			.map(|ev| ev.serialized_length())
+			.sum();
+		if total_buffered_bytes >= MAX_EVENTS_BUFFER_SIZE { return }
+		pending_events.push(event);
```

Comment:
I think it's a great idea to introduce a cap on the max size of the pending_events vector. I wonder whether it be wise to go a step further and give users a heads-up if we encounter this situation. Perhaps we could consider notifying them through an Err or log instead of silently failing to push events?